test/jdk/java/lang/ClassLoader/loadLibraryUnload/LoadLibraryUnload.java

```diff
@@ -140,20 +149,27 @@ public static void main(String[] args) throws Exception {
                 .reduce(true, (i, a) -> i && a);
 
         // expect exactly 8 errors
-        Asserts.assertTrue(exceptions.size() == 8,
-                "Expected to see 8 failing threads");
+        int expectedErrorCount = (LOADER_COUNT - 1) * 2;
+        Asserts.assertTrue(exceptions.size() == expectedErrorCount,
+                "Expected to see " + expectedErrorCount + " failing threads");
 
         Asserts.assertTrue(allAreUnsatisfiedLinkError,
                 "All errors have to be UnsatisfiedLinkError");
 
-        WeakReference<Class<?>> wClass = new WeakReference<>(clazz);
-
         // release strong refs
-        clazz = null;
         threads = null;
+        canary = null;
         exceptions.clear();
-        if (!ForceGC.wait(() -> wClass.refersTo(null))) {
-            throw new RuntimeException("Class1 hasn't been GC'ed");
+
+        // Wait for the canary for each of the libraries to be GC'd
+        // before exiting the test.
+        for (int i = 0; i < LOADER_COUNT; i++) {
+            System.gc();
+            var res = refQueue.remove();
```

Comment:
This should block until one becomes available or `InterruptedException` gets thrown.   So `res` would never be null?